syntax = "proto3";
option java_multiple_files = true;
package org.onap.ccsdk.cds.controllerblueprints.common.api;

message CommonHeader {
  string timestamp = 1;
  string originatorId = 23;
  string requestId = 3;
  string subRequestId = 4;
  Flag flag = 5;
}

message Flag {
  bool isForce = 1;
  int32 ttl = 2;
}

message ActionIdentifiers {
  string blueprintName = 1;
  string blueprintVersion = 2;
  string actionName = 3;
  string mode = 4;
}

message Status {
  int32 code = 1;
  // present only if message is failure
  string errorMessage = 2;
  // This will be success or failure
  string message = 3;
  EventType eventType = 4;
  string timestamp = 5;
}

enum EventType {
  EVENT_COMPONENT_FAILURE = 0;
  EVENT_COMPONENT_PROCESSING = 1;
  EVENT_COMPONENT_NOTIFICATION = 2;
  EVENT_COMPONENT_EXECUTED = 3;
  EVENT_COMPONENT_TRACE = 4;
}